### PR TITLE
add lambda expression example and md file

### PR DESCRIPTION
**Pull request by BMPark-Lab**
2 cpp files and 1 md file

### CODE REVIEW

**Review by astrophenomenon — APPROVED** [review]
Looks good!

File: cpp1st/week01/byongmin/lambdaExpression1.cpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+    auto addtwointegers = [](int x) -> function<int(int)> {
+        return [=](int y) { return x + y; };
+    };
+
+    auto higherorder = [](const function<int(int)>& f, int z) {
+        return f(z) * 2;
+    };
+
+    auto answer = higherorder(addtwointegers(7), 8);
```
Comment (hmlearning) [inline cpp1st/week01/byongmin/lambdaExpression1.cpp:114]: hmm. it looks complicate and a little bit hard to understand what you try. why don't you pass addtwointegers, 7 and 8?

**Review by hmlearning — APPROVED** [review]
Nice work.

File: cpp1st/week01/byongmin/lambdaExpression1.cpp (hunk 1 above)
Comment (BMPark-Lab) [reply cpp1st/week01/byongmin/lambdaExpression1.cpp:114]: The definition of higher-order function is easy, but I don't understand fully why it's useful.
Need more understanding.

**Pull request merged**